spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue595Test.java

```diff
@@ -0,0 +1,32 @@
+package edu.umd.cs.findbugs.detect;
+
+import static edu.umd.cs.findbugs.test.CountMatcher.containsExactly;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import edu.umd.cs.findbugs.AbstractIntegrationTest;
+import edu.umd.cs.findbugs.test.SpotBugsRule;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcher;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
+
+public class Issue595Test extends AbstractIntegrationTest {
+    @Rule
+    public SpotBugsRule spotbugs = new SpotBugsRule();
+
+    /**
+     * @see <a href="https://github.com/spotbugs/spotbugs/issues/595">GitHub
+     *      issue</a>
+     */
+    @Test
+    public void testRangeArrayLength() {
+        performAnalysis("rangeArray/RangeArrayLengthAndOffsetBugExample.class");
+
+        BugInstanceMatcher bugTypeMatcherLength = new BugInstanceMatcherBuilder().bugType("RANGE_ARRAY_LENGTH").build();
+        BugInstanceMatcher bugTypeMatcherOffset = new BugInstanceMatcherBuilder().bugType("RANGE_ARRAY_OFFSET").build();
+
+        assertThat(getBugCollection(), containsExactly(4, bugTypeMatcherOffset));
+        assertThat(getBugCollection(), containsExactly(13, bugTypeMatcherLength));
```

Comment:
Are you sure that these all of 4 & 13 bugs are located at expected lines?

Reply:
If not, it would be better to separate example class and test method

Reply:
Example class lists the expected lines. Example:
```java
public void ioOperation5() {
    File file = new File("temp.txt");
    try (FileReader fr = new FileReader(file)) {
        char[] buffer = new char[BUFFER_LENGTH];
        // Expected RANGE_ARRAY_OFFSET #1
        // Expected RANGE_ARRAY_LENGHT #4
        fr.read(buffer, BUFFER_LENGTH + 1, 1);
    } catch (Exception e) {
    }
}
```
I did in this way because both rules are quite related and the bug affected both of them. If you prefer I can make an independent RANGE_ARRAY_LENGHT test class.

Reply:
Ya I know meaning of these comments, I just want to ensure that this comment matches actual analysis result or not.

Reply:
I've rewritten the test class. Now it only checks one expected rule per method. I think it's clearer now.